packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/watch-blocks/watch-blocks-v1-endpoint.ts

```diff
@@ -0,0 +1,329 @@
+   *
+   * @param blockEvent filtered block
+   *
+   * @returns Nothing.
+   */
+  private monitorFilteredCallback(blockEvent: BlockEvent) {
+    const { socket, log } = this;
+    const clientId = socket.id;
+    log.debug(
+      `Filtered BlockEvent received: #${blockEvent.blockNumber.toString()}, client: ${clientId}`,
+    );
+
+    if (!("filtered_transactions" in blockEvent.blockData)) {
+      log.error("Wrong blockEvent type received - should not happen!");
+      return;
+    }
+
+    socket.emit(WatchBlocksV1.Next, {
+      filteredBlock: blockEvent,
+    });
+  }
+
+  /**
+   * Callback executed when receiving block with standard type "private"
+   * Sends WatchBlocksV1.Next with new block to the client.
+   *
+   * @param blockEvent private block
+   *
+   * @returns Nothing.
+   */
+  private monitorPrivateCallback(blockEvent: BlockEvent) {
+    const { socket, log } = this;
+    const clientId = socket.id;
+    log.debug(
+      `Private BlockEvent received: #${blockEvent.blockNumber.toString()}, client: ${clientId}`,
+    );
+
+    if (!("data" in blockEvent.blockData)) {
+      log.error("Wrong blockEvent type received - should not happen!");
+      return;
+    }
+
+    socket.emit(WatchBlocksV1.Next, {
+      privateBlock: blockEvent,
+    });
+  }
+
+  /**
+   * Get block listener callback and listener type it's expect.
+   * Returns separate function object each time it's called (this is required y fabric node SDK).
+   *
+   * @param type requested listener type (including custom Cactus ones).
+   *
+   * @returns listener: BlockListener;
+   * @returns listenerType: BlockType;
+   */
+  private getBlockListener(type: WatchBlocksListenerTypeV1) {
+    let listener: BlockListener;
+    let listenerType: EventType;
+
+    switch (type) {
+      case WatchBlocksListenerTypeV1.Full:
+        listener = async (blockEvent) => this.monitorFullCallback(blockEvent);
+        listenerType = "full";
+        break;
+      case WatchBlocksListenerTypeV1.Filtered:
+        listener = async (blockEvent) =>
+          this.monitorFilteredCallback(blockEvent);
+        listenerType = "filtered";
+        break;
+      case WatchBlocksListenerTypeV1.Private:
+        listener = async (blockEvent) =>
+          this.monitorPrivateCallback(blockEvent);
+        listenerType = "private";
+        break;
+      case WatchBlocksListenerTypeV1.CactusTransactions:
+        listener = async (blockEvent) =>
+          this.monitorCactusTransactionsCallback(blockEvent);
+        listenerType = "full";
+        break;
+      default:
+        // Will not compile if any type was not handled by above switch.
+        const unknownType: never = type;
+        throw new Error(
+          `Unknown block listen type - '${unknownType}'. Check name and connector version.`,
+        );
+    }
+
+    if (!listener || !listenerType) {
+      // Should never happen
+      throw new Error("Could not determine listener or listenerType.");
+    }
```

Comment:
@outSH I agree that this will never happen because it wouldn't compile if you didn't assign both of those variables, so IMO this can be removed. With that said if you'd still rather keep this in here (which I can also understand) then I'd make sure to include the `type` in this exception message as well so that people investigating it have more information.

Reply:
@petermetz Wow, I don't remember why it's here :D I think this is leftover from partial implementation where "unknown" type was popping out and I wanted a way to narrow it before return, but it is redundant here. Removed.

Reply:
@outSH Thank you for the update and no worries, I forget stuff like this in all the time too. Better to be paranoid with validation haha